### PR TITLE
Use different containers for testing leap and sle

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

Use different containers for testing uyuni and head.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed
- [X] **DONE**

## Test coverage
- No test
- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/14922

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 1]
```diff
@@ -1,7 +1,25 @@
 #! /bin/sh
+if [ -z ${PRODUCT+x} ];then
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:2]: What is this doing? Setting the environment variable if it doesn't exist? Why the `+x`?

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 2]
```diff
@@ -1,7 +1,25 @@
...
+    PRODUCT="Uyuni"
+fi
+
+while getopts 'P:h' c
+do
+    case $c in
+        P) PRODUCT=$OPTARG ;;
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:9]:
```suggestion
        P) VPRODUCT="VERSION.${OPTARG}";;
```

If we ever change the filenames, this makes things easier. No need to change a lot of places in the script.

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 3]
```diff
@@ -1,7 +1,25 @@
 #! /bin/sh
+if [ -z ${PRODUCT+x} ];then
+    PRODUCT="Uyuni"
+fi
+
+while getopts 'P:h' c
+do
+    case $c in
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:8]:
```suggestion
    case ${option} in
```

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 4]
```diff
@@ -1,7 +1,25 @@
 #! /bin/sh
+if [ -z ${PRODUCT+x} ];then
+    PRODUCT="Uyuni"
+fi
+
+while getopts 'P:h' c
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:6]:
```suggestion
while getopts 'P:h' option
```
Maybe more readable name for the variable?

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 5]
```diff
@@ -1,7 +1,25 @@
...
+do
+    case $c in
+        P) PRODUCT=$OPTARG ;;
+        h) echo "Usage $0 [-p PRODUCT]";exit -2;;
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:10]:
```suggestion
        h) echo "Usage ${SCRIPT} [-P PRODUCT]"; exit 2;;
```

Use unsigned integers for exit:

> NAME
>        exit — cause the shell to exit
> 
> SYNOPSIS
>        exit [n]
>
> DESCRIPTION
>       The exit utility shall cause the shell to exit with the exit status specified by the unsigned decimal integer n.  If n is specified, but its value is not between 0 and 255 inclusively, the exit status is undefined.

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 6]
```diff
@@ -1,7 +1,25 @@
...
+    esac
+done
 
 HERE=`dirname $0`
-. $HERE/VERSION
+
+if [ ! -f $HERE/VERSION.${PRODUCT} ];then
+   echo "VERSION.${PRODUCT} does not exist"
+   exit -3
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:18]:
```suggestion
   exit 3
```

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 7]
```diff
@@ -1,7 +1,25 @@
...
+    case $c in
+        P) PRODUCT=$OPTARG ;;
+        h) echo "Usage $0 [-p PRODUCT]";exit -2;;
+    esac
+done
 
 HERE=`dirname $0`
-. $HERE/VERSION
+
+if [ ! -f $HERE/VERSION.${PRODUCT} ];then
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:16]:
```suggestion
if [ ! -f ${HERE}/${VPRODUCT} ];then
```

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 8]
```diff
@@ -1,7 +1,25 @@
...
+   echo "VERSION.${PRODUCT} does not exist"
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:17]:
```suggestion
   echo "$}VPRODUCT} does not exist!"
```

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 9]
```diff
@@ -1,7 +1,25 @@
...
+   exit -3
+fi
+
+echo "Loading VERSION.${PRODUCT}"
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:21]:
```suggestion
echo "Loading ${VPRODUCT}"
```

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 10]
```diff
@@ -1,7 +1,25 @@
...
+. $HERE/VERSION.${PRODUCT}
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:22]:
```suggestion
. $HERE/${VPRODUCT}
```

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 11]
```diff
@@ -1,7 +1,25 @@
 #! /bin/sh
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:1]:
```suggestion
#! /bin/sh

SCRIPT=$(basename ${0})
```
So when we show the help, we don't show the full path that includes the PWD

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh [hunk 12]
```diff
@@ -1,7 +1,25 @@
...
-. $HERE/VERSION
+
+if [ ! -f $HERE/VERSION.${PRODUCT} ];then
+   echo "VERSION.${PRODUCT} does not exist"
+   exit -3
+fi
+
+echo "Loading VERSION.${PRODUCT}"
+. $HERE/VERSION.${PRODUCT}
 GITROOT=`readlink -f $HERE/../../../`
```
Comment (juliogonzalez) [inline susemanager-utils/testing/automation/backend-unittest-pgsql.sh:23]:
```suggestion
GITROOT=`readlink -f ${HERE}/../../../`
```
You are not changing it here, but you are changing it at the other files, so we can as well fix it everywhere :-)

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh (hunk 7 above)
Comment (juliogonzalez) [reply susemanager-utils/testing/automation/backend-unittest-pgsql.sh:16]: Enclising variable names with `{}` is always a good practice, as then you can remember to do it when the variable is inside texts.

**Review by juliogonzalez — COMMENTED** [review]
I agree with the idea. Don't see a problem about having both env variables and parameters, and in this PR the parameter takes precedence.

I left a bunch of suggestions and comments on the first scripts, that can be applied to almost all other scripts.

File: susemanager-utils/testing/automation/backend-unittest-pgsql.sh (hunk 1 above)
Comment (jordimassaguerpla) [reply susemanager-utils/testing/automation/backend-unittest-pgsql.sh:4]: https://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash

**Pull request merged**